src/plugins/visualizations/public/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -136,6 +136,7 @@ export class VisualizeEmbeddableFactory
       showNewVisModal({
         originatingApp: await this.getCurrentAppId(),
         outsideVisualizeApp: true,
+        createByValue: true,
```

Comment:
why do we need to pass this one down ? (was not needed for visualize by value ?)

Reply:
also, should we not go thru visualize embeddable to create a lens embeddable from dashboard ? i would prefer if we would directly work with lens embeddable there.

Reply:
The createByValue argument was used to differentiate between editing a by value embeddable, creating a by reference embeddable, and creating a by value embeddable. I was able to remove this prop by checking for the presence of the embeddableId in the stateTransferService instead, which should make it a little cleaner.

In terms of creating a lens embeddable via the visualize embeddable factory, it's been that way for a while due to the lens app being registered as an alias App in visualize. Changing this flow will be a larger initiative, but we're already making progress on it. @ryankeairns has a really nice design for a new 'add panel' flow.